### PR TITLE
Fixes in Doxygen documentation:

### PR DESCRIPTION
**Pull request by ihrivnac**
- Restored the main page and fixed links to internal pages
- Fixed links to sub pages
- Updated DoxygenInstructions.md
- Updated instructions in .travis.yaml and moved the filtering instructions
  (now obsolete) in commented lines in doc/scripts/filter_for_doxygen.sh

### CODE REVIEW

**Comment by ihrivnac** [comment]
@ktf, @MichaelLettrich ,
Could you, please, take a look in the mods in travis which fails now? Thank you,

**Comment by ktf** [comment]
Does it work if you revert the change you did at:

https://github.com/AliceO2Group/AliceO2/pull/2725/files#diff-354f30a63fb0907d4ad57269548329e3L68-L77

?

**Comment by ihrivnac** [comment]
@sawenzel , @Barthelemy , @davidrohr ,
Could someone review this PR? Thank you,

File: .travis.yml [hunk 1]
```diff
@@ -71,8 +71,8 @@ matrix:
         # This is required because of differences between doxygen and github
         # markdown syntax for code blocks. The committed code should use
         # the github flavour and we generate the doxygen one on the fly.
-        git grep -l '^```[a-zA-Z]' | xargs sed -i .old -e 's|```\([a-zA-Z][a-zA-Z]*\)|\n```{.\1}\n|g;s|[.]bash|.sh|g;s|```|~~~~~~~|g'
-        find . -name "*.old" -delete
+        #git grep -l '^```[a-zA-Z]' | xargs sed -i .old -e 's|```\([a-zA-Z][a-zA-Z]*\)|\n```{.\1}\n|g;s|[.]bash|.sh|g;s|```|~~~~~~~|g'
```
Comment (ktf) [inline .travis.yml]: Please do not comment code. Just remove it together with the old comment.

**Comment by ihrivnac** [comment]
@ktf , 
I removed the commented lines; is there a way to see the generated documentation before the PR is accepted? Thank you.

**Comment by ktf** [comment]
I am afraid, not.

**Comment by sawenzel** [comment]
Merging. We can re-iterate from here.

**Pull request merged**